### DISCUSSION TITLE
How to achieve per batch count randomness for immediately evaluated variables

### DISCUSSION THREAD

**Discussion by jchadwic** [adieyal/sd-dynamic-prompts#680] [Q&A]
Hi!

Is it possible to get a different value loaded into a variable on each generation for batch COUNTS? For example:

I am generating a set of images (in A1111 using the extension), and the batch count is set to 4, what I want to happen is the ability to have a variable retain the same value across multiple uses in a single generation, but select from a different potential value in the next generation.

So, something like:

${season=!{summer|autumn|winter|spring}}
In ${season}, I wear ${season} shirts and ${season} trousers

Will result in "season" using the same value in a single image, "summer" for instance, but taking on a new value each time a new image is generated.

Hopefully my question is clear, thanks in advance for any help.

**Comment by themasterswordmaster** [comment]
I as well would like to know if this is possible, Kinda annoying to make matching outfits that are different between different gens in a batch. It works if you do "Generate Forever", as thats not working with batches but just individual gens that restart infinitely. Though of course, that makes it infinite, not a set amount
